rct229/rules/section5/section5rule45.py

```diff
@@ -0,0 +1,75 @@
+from rct229.rule_engine.rule_base import RuleDefinitionListIndexedBase, RuleDefinitionBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.compare_schedules import (compare_schedules, LEAP_YEAR_HOURS, REGULAR_YEAR_HOURS)
+
+
+class Section5Rule45(RuleDefinitionListIndexedBase):
+    """Rule 45 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule45, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section5Rule45.ZoneRule(),
+            index_rmr="baseline",
+            id="5-45",
+            description="The infiltration schedules are the same in the proposed RMR as in the baseline RMR.",
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*].zones[*]",
+            required_fields={
+                "$": ["calendar"],
+                "calendar": ["is_leap_year"]
+            }
+        )
+
+    def create_data(self, context, data=None):
+        rmr_proposed = context.proposed
+        return {"is_leap_year": rmr_proposed["calendar"]["is_leap_year"]}
+
+    def list_filter(self, context_item, data=None):
+        # baseline should match to proposed
+        zone_p = context_item.proposed
+        return (
+                zone_p.get("infiltration") is not None
```

Comment:
There are a couple of more direct, pythonic ways of making this check:
1. `"infiltration" in zone_p`
2. `zone_p.has_key("infiltration")`

In this case I would favor the second option because it is clearer to a non-python programer.